### PR TITLE
Fix most linting issues in pkg/util/mount

### PR DESCRIPTION
**Pull request by codenrhoden**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
This patch adds comments to exported items that were missing them in
order to make the linter happy. Only code changes that were limited to
the scope of this package were made. There are other linting issues that
will effect callers, and that will be done a separate patch.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:
@msau42 This also addresses a portion of @thockin 's feedback in https://github.com/kubernetes/utils/pull/100. It's going to be several PRs to address all of it, but I'm hoping to do it in small chunks for easy review.

/assign @msau42 

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: pkg/util/mount/mount.go [hunk 1]
```diff
@@ -97,40 +104,22 @@ type HostUtils interface {
 	GetMode(pathname string) (os.FileMode, error)
 }
 
-type Subpath struct {
```
Comment (codenrhoden) [inline pkg/util/mount/mount.go]: This was mistakenly leftover from when `subpath` was moved into `pkg/volume/util`. It is unused here.

**Comment by neolit123** [comment]
LGTM
/sig storage
/priority backlog

**Comment by jsafrane** [comment]
Can you please remove `pkg/util/mount` from `hack/.golint_failures` to see if this PR helps and prevent future regressions?

**Comment by codenrhoden** [comment]
> Can you please remove `pkg/util/mount` from `hack/.golint_failures` to see if this PR helps and prevent future regressions?

@jsafrane Not quite yet -- the linter won't pass w/o this PR and https://github.com/kubernetes/kubernetes/pull/81186. Which ever one goes in first, I'll update the other to remove that line.

File: pkg/util/mount/mount.go [hunk 2]
```diff
@@ -61,32 +72,28 @@ type Interface interface {
...
 type HostUtils interface {
-	// DeviceOpened determines if the device is in use elsewhere
+	// DeviceOpened determines if the device (e.g. /dev/sdc) is in use elsewhere
 	// on the system, i.e. still mounted.
 	DeviceOpened(pathname string) (bool, error)
 	// PathIsDevice determines if a path is a device.
 	PathIsDevice(pathname string) (bool, error)
 	// GetDeviceNameFromMount finds the device name by checking the mount path
-	// to get the global mount path within its plugin directory
+	// to get the global mount path within its plugin directory.
```
[the remark this answers is not in the document]
Comment (codenrhoden) [reply pkg/util/mount/mount.go]: Agreed. There are a few of them like that. I haven't tried to address comments like [this](https://github.com/kubernetes/utils/pull/100/files#r311321310) yet. :/ Do you want me to try and do that here?

**Comment by jsafrane** [comment]
/approve
I let @msau42 to decide if she wants to fix misleading comments in this PR.

File: pkg/util/mount/mount.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (msau42) [reply pkg/util/mount/mount.go]: It can be done as a followup

**Pull request merged**